### PR TITLE
new example for ut:test_autoestimator_pytorch

### PR DESCRIPTION
**Pull request by liangs6212**
completed create and Copy some existing functions

### CODE REVIEW

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                                        optimizer=specify_optimizer,
+                                        loss=specify_loss_function,
+                                        logs_dir="/tmp/zoo_automl_logs",
+                                        resources_per_trial={"cpu": 2},
+                                        name="test_fit")
+    data = get_train_val_data()
+    auto_est.fit(data, recipe=LinearRecipe(), metric="accuracy")
+    # Choose the best model
+    best_model = auto_est.get_best_model()
+    best_model_mse = best_model.evaluate(x=data.get('val_x'), y=data.get('val_y'), metrics=['mse'], multioutput="raw_values")
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:109]: You shouldn't evaluate with 'mse' since it is not a regression task and the metric we used in fit is "accuracy".
Let's use "accuracy" instead.

Comment (yushan111) [reply pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:109]: x=data['val_x'], y=data["val_y"]

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    print(f'mse is {best_model_mse[0][0]}')
+    return best_model
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:111]: We may not need return

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        y = np.concatenate([y1, y2], axis=0)
+        return x, y
+    train_x, train_y = get_x_y(size=1000)
+    val_x, val_y = get_x_y(size=400)
+    data = {'x': train_x, 'y': train_y, 'val_x': val_x, 'val_y': val_y}
+    return data
+
+
+class LinearRecipe(Recipe):
+    def search_space(self, all_available_features):
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:81]: remove `all_available_features`

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+        y2 = np.ones((size // 2, 1))
+        y = np.concatenate([y1, y2], axis=0)
+        return x, y
+    train_x, train_y = get_x_y(size=1000)
+    val_x, val_y = get_x_y(size=400)
+    data = {'x': train_x, 'y': train_y, 'val_x': val_x, 'val_y': val_y}
+    return data
+
+
+class LinearRecipe(Recipe):
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:80]:
```python
def __init__(self, training_iteration, num_samples):
      self.training_iteration = training_iteration
      self.num_samples = num_samples

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+            "fc2_size": hp.choice([100, 128]),
+            LR_NAME: hp.choice([0.001, 0.003, 0.01]),
+            "batch_size": hp.choice([32, 64])
+        }
+
+    def runtime_params(self):
+        return {
+            "training_iteration": args.specify_training_iteration,
+            "num_samples": args.specify_num_samples
+        }
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:95]: remove that

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+                             "You can change it with your k8s image.")
+    parser.add_argument('--k8s_driver_host', type=str, default="",
+                        help="The k8s driver localhost. ")
+    parser.add_argument('--k8s_driver_port', type=str, default="",
+                        help="The k8s driver port.")
+    
+    parser.add_argument('-so','--specify_optimizer', type=str, default="Adam",
+                        help="Specify optimizer.")
+    parser.add_argument('-slf','--specify_loss_function', type=str, default="BCELoss",
+                        help="Specify loss function.")
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:145]: It might be a little complicated.

We could remove that from parser. And put the default values directly in `train_example`

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+                        help="The k8s driver port.")
+    
+    parser.add_argument('-so','--specify_optimizer', type=str, default="Adam",
+                        help="Specify optimizer.")
+    parser.add_argument('-slf','--specify_loss_function', type=str, default="BCELoss",
+                        help="Specify loss function.")    
+    parser.add_argument('-sti','--specify_training_iteration', type=int, default=1,
+                        help="Users can specify parameters by themselves.")
+    parser.add_argument('-sns','--specify_num_samples', type=int, default=4,
+                        help="The number of number sample.")
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:149]: '--epochs', type=int, default=1,
help="The number of epochs in each trial"
'--trials', type=int, default=4,
help="The number of searching trials"

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+            "num_samples": args.specify_num_samples
+        }
+
+
+def train_example(specify_optimizer, specify_loss_function):
+    auto_est = AutoEstimator.from_torch(model_creator=model_creator,
+                                        optimizer=specify_optimizer,
+                                        loss=specify_loss_function,
+                                        logs_dir="/tmp/zoo_automl_logs",
+                                        resources_per_trial={"cpu": 2},
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:103]: You could add this to argparse.
--cpus_per_trial,
help="The number of cores you want to allocate for each trial"

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+        }
+
+    def runtime_params(self):
+        return {
+            "training_iteration": args.specify_training_iteration,
+            "num_samples": args.specify_num_samples
+        }
+
+
+def train_example(specify_optimizer, specify_loss_function):
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:98]: def train_example(args)

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+    auto_est = AutoEstimator.from_torch(model_creator=model_creator,
+                                        optimizer=specify_optimizer,
+                                        loss=specify_loss_function,
+                                        logs_dir="/tmp/zoo_automl_logs",
+                                        resources_per_trial={"cpu": 2},
+                                        name="test_fit")
+    data = get_train_val_data()
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:105]: recipe = LinearRecipe(training_iteration=args.epochs, num_samples=args.trials)

**Comment by yushan111** [comment]
Auto test for the example is needed. You can add in https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/examples/run-example-tests-pip-ray.sh and https://github.com/intel-analytics/analytics-zoo/blob/master/pyzoo/zoo/examples/run-example-test-ray.sh

File: pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+                                        logs_dir="/tmp/zoo_automl_logs",
+                                        resources_per_trial={"cpu": args.cpus_per_trial},
+                                        name="test_fit")
+    data = get_train_val_data()
+    recipe=LinearRecipe(training_iteration=args.epochs,
+                        num_samples=args.trials)
+    auto_est.fit(data, recipe=recipe, metric="accuracy")
+    # Choose the best model
+    best_model = auto_est.get_best_model()
+    best_model_accuracy = best_model.evaluate(x=data['val_x'], y=data['val_y'], metrics=['accuracy'], multioutput="raw_values")
```
Comment (yushan111) [inline pyzoo/zoo/examples/orca/automl/autoestimator/autoestimator_pytorch.py:111]: remove `multioutput="raw_values"`

**Comment by liangs6212** [comment]
00598fc:Comply with pep8 specification and fix the bug of app_test
